internal/client/controlplane/client.go

```diff
@@ -73,3 +118,63 @@ func StatusFromResponse(resp *controlplanes.ControlPlaneResponse) v1alpha1.Contr
 
 	return status
 }
+
```

Comment:
Since packages use SemVar, this might be a bit better https://pkg.go.dev/golang.org/x/mod/semver than having your own custom parser. That said, I wouldn't block on that.

Reply:
Crossplane uses https://github.com/Masterminds/semver. We should likely use that library here. The semantics of those libraries differs unfortunatelly in little details.

Reply:
the problem here is that everything after + is metadata and not compared by default libraries - i changed my PR that i use the library for `minor major patch` compare and use a `strings.Compare` for the metadata